services/thermal_mgr/thermal_mgr.c

```diff
@@ -44,18 +45,71 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
   /* Send an event to the thermal manager queue */
 
-  return ERR_CODE_SUCCESS;
+  if (event == NULL)
+  {
+    return ERR_CODE_INVALID_ARG; 
+  }
+
+  if (xQueueSend(thermalMgrQueueHandle, (void *) event, 0) == pdTRUE)
```

Comment:
nit: I would check if `thermalMgrQueueHandle == NULL` (this could occur if the queue hasn't been created yet) before sending to the queue.